master/internal/proxy/proxy.go

```diff
@@ -126,9 +121,12 @@ func (p *Proxy) newProxyHandler(serviceID string) echo.HandlerFunc {
 
 		// Proxy the request to the target host.
 		var proxy http.Handler
-		if c.IsWebSocket() {
+		switch {
+		case strings.HasSuffix(c.Request().URL.Path, "tcp"):
```

Comment:
It seems wrong to let the client decide if the websocket is a tunnel or not.  I think we should definitely just store this information in the Service struct.

Reply:
I'm not quite so sure, but I've changed it. (From an external perspective, it seems more weird to me to select between incompatible wire protocols based on the opaque UUID rather than a visible URL.)

Reply:
I can't argue with that, but since for every uuid you need to know a priori whether or not to include the /tcp suffix it feels the same. Arguably the issue is passing everything through the same proxy/ route.